azure-quantum/tests/unit/test_plugins.py

```diff
@@ -0,0 +1,294 @@
+import unittest
+import warnings
+
+from qiskit import QuantumCircuit
+from cirq import ParamResolver
+
+from azure.quantum.job.job import Job
+from azure.quantum.plugins.qiskit import AzureQuantumProvider
+from azure.quantum.plugins.cirq import AzureQuantumService
+from azure.quantum.plugins.cirq.targets.target import Target
+
+from common import QuantumTestBase, ZERO_UID
+
+class TestQiskit(QuantumTestBase):
+    """TestIonq
+
+    Tests the azure.quantum.target.ionq module.
+    """
+
+    mock_create_job_id_name = "create_job_id"
+    create_job_id = Job.create_job_id
+
+    def get_test_job_id(self):
+        return ZERO_UID if self.is_playback \
+               else Job.create_job_id()
+    
+    def _3_qubit_ghz(self):
+        circuit = QuantumCircuit(3, 3)
+        circuit.name = "Qiskit Sample - 3-qubit GHZ circuit"
+        circuit.h(0)
+        circuit.cx(0, 1)
+        circuit.cx(1, 2)
+        circuit.measure([0,1,2], [0, 1, 2])
+
+        return circuit
+
+    def test_plugins_submit_qiskit_to_ionq(self):
+        circuit = self._3_qubit_ghz()
+        self._test_qiskit_submit_ionq(circuit=circuit, num_shots=None, num_shots_actual=500)
+    
+    def test_plugins_submit_qiskit_qobj_to_ionq(self):
+        from qiskit import assemble
+        circuit = self._3_qubit_ghz()
+        qobj = assemble(circuit)
+        self._test_qiskit_submit_ionq(circuit=qobj, num_shots=None, num_shots_actual=1024)
+
+    def _test_qiskit_submit_ionq(self, circuit, num_shots, num_shots_actual):
+
+        with unittest.mock.patch.object(
+            Job,
+            self.mock_create_job_id_name,
+            return_value=self.get_test_job_id(),
+        ):
+            workspace = self.create_workspace()
+            provider = AzureQuantumProvider(workspace=workspace)
+            backend = provider.get_backend("ionq.simulator")
+            
+            qiskit_job = backend.run(
+                circuit=circuit,
+                shots=num_shots
+            )
+
+            # Make sure the job is completed before fetching the results
+            # playback currently does not work for repeated calls
+            # See: https://github.com/microsoft/qdk-python/issues/118
+            if not self.is_playback:
+                job = qiskit_job._azure_job
+                self.assertEqual(False, job.has_completed())
```

Comment:
we should be testing the corresponding methods on the qiskit_job, not on the azure job.